src/prefect/engine/task_runner.py

```diff
@@ -84,11 +84,11 @@ def __init__(
         self,
         task: Task,
         state_handlers: Iterable[Callable] = None,
-        result_handler: "ResultHandler" = None,
+        result_handler: "ResultHandler" = None, # todo: here
     ):
         self.context = prefect.context.to_dict()
         self.task = task
-        self.result_handler = task.result_handler or result_handler
+        self.result_handler = task.result_handler or result_handler # todo: here
```

Comment:
Around here is where I was converting self.result_handler into a self.result on behalf of the user IF they sent me a result handler and no result.